utils/build-script

```diff
@@ -40,6 +40,47 @@ import swift_build_support.cmake
 from swift_build_support.migration import migrate_impl_args
 
 
+# Utility for selecting host compilers.
+class CompilerExecutable(object):
```

Comment:
Why not use the existing `CompilerExecutable` type instead, and turn the constructor into a global function?  Keeping the number of types low is good, IMHO.